src/twisted/internet/testing.py

```diff
@@ -967,3 +967,54 @@ def createWithCleanup(cls, testInstance: TestCase, publisher: LogPublisher) -> S
         publisher.addObserver(obs)
         testInstance.addCleanup(lambda: publisher.removeObserver(obs))
         return obs
+
+
+def benchmarkWithReactor(test_target):
+    """
+    Decorator for running a benchmark tests that loops the reactor.
+
+    This is designed to decorate test method executed using pytest and
+    pytest-benchmark.
+    """
+
+    @inlineCallbacks
+    def deferredWrapper():
+        return test_target()
```

Comment:
The overhead of starting a reactor can be high enough in some cases that the run time of the code being benchmarked will be overwhelmed, or at least muted, by that.

Two options:

1. Run the code in a loop. It's unclear how many iterations would make sense though, so perhaps there should be a default that's overrideable? (Being overrideable might be... annoying given the other recommendation I am going to give.)
2. Leave it up to the person writing the benchmark, and maybe just document this.

Reply:
We can have the reactor as a pytest session fixture.

With that, we start the reactor only once for all the pytest tests.

Reply:
Looking at timing locally, the setup cost of the SSH test appears to overwhelm both the reactor startup/shutdown and, more importantly, the actual thing nominally being benchmarked (the handshake). Possibly I can solve this with passing in the `benchmark` object. Maybe.

Reply:
Ah, no. I broke _something_ and the benchmark is starting, but never finishing at all :facepalm: So it's not measuring what it ought to be measuring.

Reply:
OK, and having fixed the issue and the underlying issue, I think the ssh benchmark at least is slow enough that one instance is sufficient. Which suggests looping should be in benchmarks, not in the framework.